src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs

```diff
@@ -107,14 +108,35 @@ public AuthenticationContext(string authority, bool validateAuthority, TokenCach
         {
         }
 
+        /// <summary>
+        /// Constructor to create the context with the address of the authority and flag to turn address validation off.
+        /// Using this constructor, address validation can be turned off. Make sure you are aware of the security implication of not validating the address.
+        /// 
+        /// This constructor allows the HttpClient to be provided instead of having a new one created.  This can be useful when working with multiple different tenants.
+        /// </summary>
+        /// <param name="authority">Address of the authority to issue token.</param>
+        /// <param name="validateAuthority">Flag to turn address validation ON or OFF.</param>
+        /// <param name="tokenCache">Token cache used to lookup cached tokens on calls to AcquireToken</param>
+        public AuthenticationContext(string authority, bool validateAuthority, TokenCache tokenCache, HttpClient httpClient)
+            : this(null, authority, validateAuthority ? AuthorityValidationType.True : AuthorityValidationType.False,
+                   tokenCache, httpClient)
+        {
+        }
+
         internal AuthenticationContext(IServiceBundle serviceBundle, string authority, AuthorityValidationType validateAuthority,
-            TokenCache tokenCache)
+            TokenCache tokenCache, HttpClient httpClient = null)
         {
             // If authorityType is not provided (via first constructor), we validate by default (except for ASG and Office tenants).
             Authenticator = new Authenticator(authority, (validateAuthority != AuthorityValidationType.False));
             TokenCache = tokenCache;
 
-            _serviceBundle = serviceBundle ?? ServiceBundle.CreateDefault();
```

Comment:
>_serviceBundle = serviceBundle ?? ServiceBundle.CreateDefault(); [](start = 12, length = 64)

IMO: it would be more clean to call ServiceBundle.CreateWithHttpClient(), as we are handling the object creation in the ServiceBundle class and not here.  Also you have to CreateDefault if httpClient is null, this code can't tell if that is what is happening or not.